File: Documentation/5.4/Samples/python/ClientApi/Operations/Maintenance/Indexes/ResetIndex.py
from ravendb.documents.operations.definitions import VoidMaintenanceOperation
from ravendb.documents.operations.indexes import ResetIndexOperation

from examples_base import ExampleBase


class ResetIndex(ExampleBase):
    def setUp(self):
        super().setUp()

    def test_reset_index(self):
        with self.embedded_server.get_document_store("ResetIndex") as store:
            # region reset
            # Define the reset index operation, pass index name
            reset_index_op = ResetIndexOperation("Orders/Totals")

            # Execute the operation by passing it to maintenance.send
            # An exception will be thrown if index does not exist
            store.operations.send(reset_index_op)
            # endregion


class Foo:
    # region syntax
    class ResetIndexOperation(VoidMaintenanceOperation):
        def __init__(self, index_name: str): ...

    # endregion

File: Documentation/5.4/Samples/php/ClientApi/Operations/Maintenance/Indexes/ResetIndex.php
<?php

namespace RavenDB\Samples\ClientApi\Operations\Maintenance\Indexes;

use RavenDB\Documents\DocumentStore;
use RavenDB\Documents\Operations\Indexes\ResetIndexOperation;

class ResetIndex
{
    public function samples(): void
    {
        $store = new DocumentStore();
        try {
            # region reset
            // Define the reset index operation, pass index name
            $resetIndexOp = new ResetIndexOperation("Orders/Totals");

            // Execute the operation by passing it to Maintenance.SendAsync
            // An exception will be thrown if index does not exist
            $store->maintenance()->send($resetIndexOp);
            # endregion

        } finally {
            $store->close();
        }
    }
}

/*
interface IFoo
{
    # region syntax
    public ResetIndexOperation(?string $indexName);
    # endregion
}
*/
